clang/lib/Headers/aiev2_upd_ext.h

```diff
@@ -527,9 +527,9 @@ INTRINSIC(v32bfloat16) insert(v32bfloat16 a, int idx, v16bfloat16 b) {
 // Set 256-bit portion of 512-bit register
 INTRINSIC(v32bfloat16) set_v32bfloat16(int idx, v16bfloat16 b) {
   if (idx == 0)
-    return __builtin_aiev2_set_bf512_bf256(b, 0);
+    return __builtin_aiev2_set_I512_I256(b, 0);
```

Comment:
This looks curious, all values different from 0 can be used as index `1`. I was wondering why not simply:
```
return __builtin_aiev2_set_I512_I256(b, idx == 0 ? 0 : 1);
```

Reply:
I think those headers were partly auto-generated, and for some intrinsics we have more than two different values. So I'd assume it's easier to write a generic generator if sticking to `if`/`else`.